src/App.tsx

```diff
@@ -1,61 +1,125 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  // eslint-disable-next-line no-console
+  console.log(foundUser, userId);
+
+  return foundUser || null;
+}
+
+const combinedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  userId: getUser(todo.userId),
+}));
 
 export const App = () => {
+  const [todos, addTodos] = useState(combinedTodos);
+  const [inputText, setInputText] = useState('');
+  const [inputUserId, setInputUserId] = useState(-1);
+  const [isValid, setIsValid] = useState(true);
+
+  let largestId = todos.reduce((prev, curr) => (
+    prev > curr.id
+      ? prev
+      : curr.id
+  ), 0);
+
+  function onSubmit(todoText: string, userId: number) {
+    if (userId === -1 || todoText.trim() === '') {
+      setIsValid(false);
+
+      return;
+    }
+
+    setIsValid(true);
+    largestId += 1;
+
+    const newTodo : Todo = {
+      id: largestId,
+      title: todoText,
+      completed: false,
+      userId: getUser(userId),
+    };
+
+    addTodos([...todos, newTodo]);
+
+    setInputText('');
+    setInputUserId(-1);
+  }
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+          onSubmit(inputText, inputUserId);
```

Comment:
you use ```onSubmit``` and ```preventDefault``` in ```onSubmit```
- follow [these](https://medium.com/javascript-in-plain-english/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364) naming conventions for methods